driver/src/driver/scheduler/system.rs

```diff
@@ -133,11 +128,69 @@ impl<'a> SystemScheduler<'a> {
     }
 }
 
-impl Scheduler for SystemScheduler<'_> {
-    fn start(&mut self) -> ! {
-        loop {
-            thread::sleep(self.run_once(SystemTime::now()));
+#[derive(Debug, Eq, PartialEq)]
+enum SolveAgain {
```

Comment:
ubernit: Semantically equivalent to an Option<Duration> but since it's private it doesn't matter much and if you think this type adds value feel free to leave like this.

Reply:
This is true. The difference is that with an option I would write a comment saying what `None` and `Some` means but with the enum it is clear from the types. On the other hand I lose convenience methods implemented on Option. In this case I prefer the enum.